internal/controller/flow_aggregation.go

```diff
@@ -0,0 +1,270 @@
+	Proto string
+	// SourceEndpoint contains k8s metadata for source endpoint
+	SourceEndpoint string
+	// DestinationEndpoint contains k8s metadata for destination endpoint
+	DestinationEndpoint string
+}
+
+type FlowCache struct {
+	cache      map[FlowKey]*list.Element
+	queue      *list.List
+	bufferSize int
+}
+
+// cacheManagerIndefinitely manages the flow cache by evicting expired flows based on the active timeout,
+// processing new flows, and resetting the timer for the next expiration.
+func (sm *streamManager) cacheManagerIndefinitely(ctx context.Context, logger *zap.Logger) error {
+	const activeTimeout = 20 * time.Second // Define the active timeout period for flows. TODO: make configurable
+
+	// Create a new timer to trigger every activeTimeout duration.
+	timer := time.NewTimer(activeTimeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done(): // If the context is canceled, exit the loop and return the error.
+			return ctx.Err()
+
+		case <-timer.C:
+			now := time.Now().UTC()
+			expiredCutoff := now.Add(-activeTimeout) // Determine the cutoff time for expired flows.
+
+			// Iterate through the flow cache queue to evict expired flows.
+			for sm.FlowCache.queue.Len() > 0 {
+				frontElem := sm.FlowCache.queue.Front()
+				var flowTimestamp time.Time
+				var flowKey FlowKey
+				var err error
+
+				// Check the type of flow and get the flow timestamp and flow key.
+				switch f := frontElem.Value.(type) {
+				case *pb.FalcoFlow:
+					flowTimestamp = f.GetTimestamp().AsTime()
+					flowKey, err = sm.createFlowKey(f)
+				case *pb.CiliumFlow:
+					flowTimestamp = f.GetTime().AsTime()
+					flowKey, err = sm.createFlowKey(f)
+				default:
+					break
+				}
+				if err != nil {
+					return err
+				}
+
+				// If the flow is not expired (timestamp is newer than the cutoff), stop evicting.
+				if flowTimestamp.After(expiredCutoff) {
+					break
+				}
+
+				// Evict the expired flow by removing it from the cache and queue.
+				sm.FlowCache.queue.Remove(frontElem)
+				delete(sm.FlowCache.cache, flowKey)
+
+				sm.sendNetworkFlowRequest(logger, frontElem.Value)
+			}
+
+			// Reset the timer to expire based on the next flow's timestamp.
+			if front := sm.FlowCache.queue.Front(); front != nil {
+				switch f := front.Value.(type) {
+				case *pb.FalcoFlow:
+					resetTimer(timer, f.GetTimestamp().AsTime(), activeTimeout)
+				case *pb.CiliumFlow:
+					resetTimer(timer, f.GetTime().AsTime(), activeTimeout)
+				default:
+					timer.Reset(activeTimeout)
+				}
+			} else {
+				// If there are no flows left, reset the timer to the default activeTimeout period.
+				timer.Reset(activeTimeout)
+			}
+
+		case flow := <-sm.flowChannel:
+			flowKey, err := sm.createFlowKey(flow)
+			if err != nil {
+				return err
+			}
+
+			// If the flow already exists in the cache, skip adding it again.
+			if _, found := sm.FlowCache.cache[flowKey]; found {
+				continue
```

Comment:
You need to increment the flow count, and potentially the byte count (which we'll get in OVN IPFIX flows).
If you don't want to do that in this PR, at the minimum add a big fat TODO comment to remember to do that later.